File: android/app/src/main/java/com/schoolfinder/app/data/Session.java
package com.schoolfinder.app.data;

/** The signed-in user. In dev-auth mode this is sent to the backend via X-Debug-* headers. */
public class Session {
    public final String uid;
    public final String displayName;
    public final String role; // "STUDENT" or "ADMIN"

    public Session(String uid, String displayName, String role) {
        this.uid = uid;
        this.displayName = displayName;
        this.role = role;
    }

    public boolean isAdmin() {
        return "ADMIN".equals(role);
    }
}

File: android/app/src/main/java/com/schoolfinder/app/data/Session.kt
package com.schoolfinder.app.data

/** The signed-in user. In dev-auth mode this is sent to the backend via X-Debug-* headers. */
data class Session(
    val uid: String,
    val displayName: String,
    val role: String, // "STUDENT" or "ADMIN"
) {
    val isAdmin: Boolean get() = role == "ADMIN"
}

/** Process-wide holder so the OkHttp interceptor can read the current session. */
object SessionHolder {
    @Volatile
    var current: Session? = null
}
